### DISCUSSION TITLE
[Feature] enable ffmpeg decoding hardware accelleration while generating video thumbnails

### DISCUSSION THREAD

**Discussion by MatLegat** [immich-app/immich#14016] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When using RKMPP full hardware transcoding on the RK3588 chip (with the `Acceleration API` option set to `RKMPP` and the `Hardware decoding` option toggle enabled), the Transcode Videos task are executed way faster, and even with the transcode job concurrency set to `4`, it still leaves half of the CPU available for other jobs and programs to run.
Before configuring it, with hardware decoding disabled and only 1 concurrent transcode task, it utilized 100% of all cores, taking a lot of time and slowing down the entire system.
That is an excellent feature, thanks to everyone involved! 😄

But... i couldn't help but notice on `htop` that ffmpeg was still sometimes being called without the `hwaccel` related flags, slowing down all jobs and other system applications. Upon further investigation, I've found that those unoptimized calls were being made by the `Generate Thumbnails` job, as it uses ffmpeg and does some kind of processing to find a good non-black frame (based on what i've found on other issues and PRs).
So basically, on my system with the RK3588 chip, it takes as much time to generate a single video thumbnail (with 2 different times) as it takes to transcode one entire 1min video from 4k to 1440p, but the thumbnail part slows everithing else running.

I've managed to reduce the impact on other jobs and apps by limiting the number of `Threads` (under `Video Transcode Settings`), and the `-threads` flag with the configured number was indeed passed to the ffmped thumbnail process.

So, the feature extension i would like is: also apply the `Hardware decoding` option to the video thumbnail generation ffmpeg process.
I have no idea on how the commands are being built/concatenated on the server code, but looks like the feature implementation should be something like copying the `hwaccel` ffmpeg flags (based on the selected accel API), replicate them on the thumbnail command if the `Hardware decoding` option is enabled, and test if they work on all of the supported accelleration APIs.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#11222

**Comment by mertalev** [comment]
Making it use HW decoding would be fairly straightforward. The filters and encoding would still have to be on CPU, though, since there isn't good support for thumbnail selection on GPU from what I've seen (the exception being CUDA with `thumbnail_cuda`). But decoding takes the most time, so it should still be a big improvement.
